scripts/funcTests/runFuncTests.sh

```diff
@@ -36,18 +38,36 @@ chmod +x scripts/funcTests/dotnet-install.sh
 
 # Get recommended version for bootstrapping testing version
 # Issue 8936 - DISABLED TEMPORARILY cli/dotnet-install.sh -i cli -c 2.2
-scripts/funcTests/dotnet-install.sh -i cli -c 2.2 -NoPath
+scripts/funcTests/dotnet-install.sh -i cli -c 2.2 -nopath
+
+if (( $? )); then
+	echo "The .NET CLI Install failed!!"
+	exit 1
+fi
+
+# Disable .NET CLI Install Lookup
+DOTNET_MULTILEVEL_LOOKUP=0
 
 DOTNET="$(pwd)/cli/dotnet"
 
+# Let the dotnet cli expand and decompress first if it's a first-run
+$DOTNET --info
 
+# Get CLI Branches for testing
 echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
-# run it twice so dotnet cli can expand and decompress without affecting the result of the target
-dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting
-DOTNET_BRANCHES="$(dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting)"
-echo $DOTNET_BRANCHES | tr ";" "\n" |  while read -r DOTNET_BRANCH
+
+IFS=$'\n'
+CMD_OUT_LINES=(`dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting`)
+# Take the line just before the last empty line and remove all the spaces
+DOTNET_BRANCHES=${CMD_OUT_LINES[-1]//[[:space:]]}
```

Comment:
Seems this is causing mac build to fail. Here's what I found from the log...

<details>
<summary>This line fails with a `bad array subscript` error.</summary>

```log
•••
dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting

/Users/runner/work/1/s/scripts/funcTests/runFuncTests.sh: line 62: CMD_OUT_LINES: bad array subscript

.NET Core SDK (reflecting any global.json):
 Version:   2.2.402
•••
.NET Core SDKs installed:
  2.2.402 [/Users/runner/work/1/s/cli/sdk]
•••
```
</details>

You are right with the `zsh`/`bash` thing on a mac.

This line takes the last line from the `GetCliBranchForTesting` target and strips out any whitespace across the line.

MSBuild already outputs with an indent, and you could also have whitespace within the string. So, you’ll have an output string like these...

1. `    release/5.0.2xx : 5.0.201`
2. `    master ; release/5.0.2xx : 5.0.201`

Which are perfectly valid but POSIX shell doesn't srtip leading/trailing spaces when assigned to a variable leading to a bunch of issues.

One of the reasons I used colon (**`:`**) as a separator to avoid splitting by spaces which could result in empty entries in the array depending on the behavior of split on each platform.

It's not like in `PowerShell` where you can trim leading/trailing whitespace using `string.Trim()` method.

In summary, we need this fix. So, I'll try a bunch of ways to strip spaces or change the shell as you suggested to see which works on both mac and linux.

If the base OS images you guys use is same as the ones from Azure DevOps, then it should be easy to repro and fix it.